src/metabase/lib/schema/expression.cljc

```diff
@@ -1,39 +1,191 @@
 (ns metabase.lib.schema.expression
   (:require
+   [malli.core :as mc]
+   [metabase.lib.schema.common :as common]
    [metabase.lib.schema.filter :as filter]
    [metabase.lib.schema.literal :as literal]
    [metabase.lib.schema.ref :as ref]
+   [metabase.lib.schema.temporal-bucketing :as temporal-bucketing]
+   [metabase.types]
    [metabase.util.malli.registry :as mr]))
 
+(comment metabase.types/keep-me)
+
+(defn- ref-of-base-type [base-type]
```

Comment:
Generates a schema for a `ref` (e.g. a `:field` clause) that has a `:base-type` that `isa?` base-type

This lets us do things like count

```clj
[:field {:base-type :type/Boolean} 1]
```

as a `::boolean` expression and strictly enforce those type checks